src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -78546,26 +78546,35 @@ class C
 {
     void M()
     {
-        var lam1 = (object o = null) => {}; // 1
+        var lam1 = (object o = null) => {};
         var lam2 = (object o = null!) => {};
-        var lam3 = (object o = default) => {}; // 2
+        var lam3 = (object o = default) => {};
         var lam4 = (object o = default!) => {};
         var lam5 = ([AllowNull] string s = null) => {};
-        var lam6 = ([DisallowNull] string? s = null) => {}; // 3
+        var lam6 = ([DisallowNull] string? s = null) => {};
         var lam7 = ([DisallowNull] string? s = "a") => {};
```

Comment:
>var lam7 = ([DisallowNull] string? s = "a") => {}; [](http://example.com/codeflow?start=8&length=50)

Consider including a synthesized delegate type. For example, perhaps change this to: `(ref int i, [DisallowNull] string? s = "a") => { }`.

Reply:
All of them are synthesized since they have optional parameters (the error messages say `<anonymous delegate>`).